src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1519,4 +1519,16 @@
     <value>and all other platforms</value>
     <comment>This call site is reachable on: 'windows' 10.0.2000 and later, and all other platforms</comment>
   </data>
+  <data name="UseStringContainsCharOverloadWithSingleCharactersDescription" xml:space="preserve">
+    <value>'string.Contains(string)' internally uses an obsolete inner loop when searching for a single character. 'string.Contains(char)' will internally use 'string.IndexOf(char)'.</value>
```

Comment:
❓ Why not update `string.Contains(string)` to use `string.IndexOf(char)` automatically? It seems like this description is prone to quickly becoming obsolete.

Reply:
I asked this question in the issue since it wasnt clear from the spec and i was wondering the same:

https://github.com/dotnet/runtime/issues/47180#issuecomment-779481857

Basically the idea is that the Analyzers are still used by older frameworks where it is unlikely that this state will change anytime soon. I agree with your reasoning though i'm not sure how to word it better. 😄

Reply:
That doesn't seem like a reason that would warrant an analyzer to fix. Are the performance problems with the current code bad enough that we _need_ to put in an analyzer as an urgent patch for users who can't update to a version which is already fixed?

Reply:
I _really_ don't like this description. `string.Contains(string)` is a well-defined and reasonably-performing method when used with a single character. It is not obsolete for this usage, and will not be obsolete for this usage in the future.

Reply:
I'll send @buyaa-n a message and try to get this resolved.

Reply:
I tend to agree with @sharwell here.

Reply:
Well seems i did not exactly suggest this message, sorry if I misguide you, i agree we shouldn't say that it is "internally uses an obsolete inner loop". My point was to mention that there is 'string.Contains(char)' overload is available which performs better for single char lookup, and that should be enough description even for experienced developers as there was no char overload before.

Reply:
I'll update the string tonight:
Does `'string.Contains(char)' is available as a better performing overload for single char lookup.`  work for y'all?

BTW: the obsolete in the original description wasnt saying the overload itself was obsolete. i was saying running that extra loop was obsolete if you know you're searching for a single char.

Reply:
> Does 'string.Contains(char)' is available as a better performing overload for single char lookup. work for y'all?

Sounds fine, another version could be `For a single char lookup the 'string.Contains(char)' overload would perform better than  'string.Contains(string)'` 

> BTW: the obsolete in the original description wasnt saying the overload itself was obsolete. i was saying running that extra loop was obsolete if you know you're searching for a single char.

Understand your intention, but for a quick glance it does sound like it has an obsolete loop 😄